Agents/SewageAgent/README.md

```diff
@@ -0,0 +1,54 @@
+# Sewerage Network Agent
+
+This agent is for maintaining data and the corresponding instances in the knowledge graph (KG) regarding the sewerage network located in a midsize town in Germany. Its purpose is to instantiate instances of the sewerage network. 
+
+## Usage 
+This part of the README describes the usage of the sewerage network agent. The module itself can be packaged into an executable war, deployed as a web servlet on tomcat. Sending the appropriate request to the correct URL will initiate the agent. 
+
+The next section explains the requirements to run the agent.
+
+### Requirements
+It is required to have access to a knowledge graph SPARQL endpoint. It can run on the same machine or need to be accessible from the host machine via a fixed URL.
+
+This can be either in form of a Docker container or natively running on a machine. It is not in the scope of this README to explain the set-up of a knowledge graph triple store.
+
+#### Building the Sewerage Network Agent
+
+Sewerage Network Agent is set up to use the Maven repository. Credentials in single-word text files located like this are required:
+```
+./credentials/
+    repo_username.txt
+    repo_password.txt
+```
+repo_username.txt should contain your github username, and repo_password.txt your github [personal access token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token),
+which must have a 'scope' that [allows you to publish and install packages](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry#authenticating-to-github-packages).
+
+To build and start the agent, open up the command prompt in the same directory as this README, run
+```
+docker-compose up -d
+```
+
+This agent is reachable at "seweragenetwork-agent/performsewageupdate" on localhost port 1080.
+
+#### Run the agent
+To run the agent, a POST request must be sent to http://localhost:1080/seweragenetwork-agent/performsewageupdate with a JSON Object. In this agent, a random JSON Object will do. Follow the sample request shown below.
+```
+
+POST http://localhost:1080/seweragenetwork-agent/performsewageupdate
+Content-Type: application/json
+{"SewerageNetworkAgent\":"DataInstantiation"}
+```
+
+In curl syntax:
+```
+curl -X POST --header "Content-Type: application/json" -d "{
+\"SewerageNetworkAgent\\":\"DataInstantiation\"}"  http://localhost:1080/seweragenetwork-agent/performsewageupdate
+```
+
+If the agent runs successfully, you should see a returned JSON Object that is similar to the one shown below.
+```
+{"Result":"Data has been instantiated."}
+```
+
+#### Import the CSV data file into the agent
```

Comment:
What does that mean? In case I "simply" want to instantiate the "original" sewage network as is
1) What would I need to do exactly? 
2) What .csv file(s) would I need to use? And why .csv(s) in the first place? Wasn't the initial data in .xml, so, where do those .csv files come from and how are they generated?

Reply:
> What does that mean? In case I "simply" want to instantiate the "original" sewage network as is
> 
> 1. What would I need to do exactly?
> 2. What .csv file(s) would I need to use? And why .csv(s) in the first place? Wasn't the initial data in .xml, so, where do those .csv files come from and how are they generated?

We converted the initial data in .xml to CSV for data treatment with a python script. CSV is used because the instantiation code was initially written to take in CSV files.  All the CSV files can be found at "Dropbox (Cambridge CARES)\IRP3 CAPRICORN shared folder\HSXUE\Sewage Instantiation Data"

The current dockerised code only takes in "HG_instances_0.csv" and "KG_instances_0.csv" files. There are 7 HG_instances and 11 KG_instances CSV files. Only need to change the stringpath names in the Dockerfile for instantiation the other CSV files. Why like this: because last minute we notice the agent cannot take in one CSV file with all the data, it will truncate the data at some where due to the column number exceeds its limit. Previously when we tested one CSV file with only German abbreviation letters for the instances, it worked fine. Now this problem arised when we replaced all the abbreviation letters with full English meaning.

Reply:
I suggest you include those python scripts into this agent, potentially into some `preprocessing` subfolder. If the original data comes in `.xml` that should be the input to the agent. Otherwise, there is no control over where things break: This could be the agent, this could be the preprocessing, this could be extraction from xml into csv, ...

Haven some "random" `.csv` file as input without documentation how it has been created feels wrong to me.

Reply:
sure, sounds good